htlcswitch/interceptable_switch.go

```diff
@@ -428,10 +483,30 @@ func (s *InterceptableSwitch) validateResolution(res *FwdResolution) error {
 	// For unencrypted messages, verify that they are parseable.
 	case !res.FailureMessageEncrypted:
 		r := bytes.NewReader(res.FailureMessage)
-		_, err := lnwire.DecodeFailureMessage(r, 0)
+		msg, err := lnwire.DecodeFailureMessage(r, 0)
 		if err != nil {
 			return ErrUnparseableFailureMsg
 		}
+
+		// Re-sign the channel update if present.
+		update := getChannelUpdateRef(msg)
+		if update != nil {
+			err := s.resignChannelUpdate(update)
+			if err != nil {
+				return err
+			}
+
+			// Re-encode the failure message for delivery to the
+			// sender.
+			var w bytes.Buffer
+
+			err = lnwire.EncodeFailureMessage(&w, msg, 0)
+			if err != nil {
+				return err
+			}
+
+			res.FailureMessage = w.Bytes()
```

Comment:
IMO this shouldn't mutate the failure message from underneath the caller. This method currently handles just validation, so should keep doing that one thing. 

If we still like this re-sign portion, then we should move it to a more specific layer.

Reply:
Agreed that this is an obscure side effect.

Separated validation and resigning. It is still mutating the resolution to avoid copying the failure message for which there doesn't seem to exist a standard method. But it is much more explicit now.